src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClientProxy.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections.Generic;
+using NLog;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Indexers.Tribler;
+
+namespace NzbDrone.Core.Download.Clients.Tribler
+{
+    public interface ITriblerDownloadClientProxy
+    {
+        ICollection<Download> GetDownloads(TriblerDownloadSettings settings);
+
+        ICollection<File> GetDownloadFiles(TriblerDownloadSettings settings, Download downloadItem);
+
+        GetTriblerSettingsResponse GetConfig(TriblerDownloadSettings settings);
+
+        void RemoveDownload(TriblerDownloadSettings settings, DownloadClientItem item, bool deleteData);
+
+        string AddFromMagnetLink(TriblerDownloadSettings settings, AddDownloadRequest downloadRequest);
+    }
+
+    public class TriblerDownloadClientProxy : ITriblerDownloadClientProxy
+    {
+        protected readonly IHttpClient _httpClient;
+        private readonly Logger _logger;
+
+        public TriblerDownloadClientProxy(IHttpClient httpClient, Logger logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        private HttpRequestBuilder getRequestBuilder(TriblerDownloadSettings settings, string relativePath = null)
+        {
+            var requestBuilder = new HttpRequestBuilder(GetBaseUrl(settings, relativePath))
+                .Accept(HttpAccept.Json);
+
+            requestBuilder.Headers.Add("X-Api-Key", settings.ApiKey);
+
+            requestBuilder.LogResponseContent = true;
+
+            return requestBuilder;
+        }
+
+        public string GetBaseUrl(TriblerDownloadSettings settings, string relativePath = null)
```

Comment:
Only used in one spot and not very long, you shouldn't need to extract it out to it's own function